File: emdb/js/emdb-common/src/model/Person.ts
import { Gender } from "./Gender.js";

export interface Person {
  id: number;
  tmdbId: number;
  name: string;
  birthDate: string | null;
  deathDate: string | null;
  gender: Gender;
  birthPlace: string | null;
  profile: string | null;
  biography: string | null;
}

File: emdb/emdb-ingest-core/src/main/java/com/erdouglass/emdb/ingest/application/dto/Person.java
package com.erdouglass.emdb.ingest.application.dto;

import java.util.Objects;
import java.util.Optional;

import com.erdouglass.emdb.media.TmdbId;

import lombok.Builder;

@Builder
public record Person(
    TmdbId tmdbId,
    String name,
    Optional<String> birthDate,
    Optional<String> deathDate,
    Optional<String> gender,
    Optional<String> biography) {

  public Person {
    Objects.requireNonNull(tmdbId, "tmdbId must not be null");
    Objects.requireNonNull(name, "name must not be null");
  }
}
